src/daemon/daemon.cpp

```diff
@@ -1968,7 +1968,17 @@ grpc::Status mp::Daemon::shutdown_vm(VirtualMachine& vm, const std::chrono::mill
     using St = VirtualMachine::State;
     const auto skip_states = {St::off, St::stopped, St::suspended};
 
-    if (std::none_of(cbegin(skip_states), cend(skip_states), [&state](const auto& st) { return state == st; }))
+    if (state == St::unknown)
+    {
+        if (delay.count() > 0)
+        {
+            return grpc::Status(grpc::StatusCode::FAILED_PRECONDITION,
+                                "Cannot delay shutdown when instance is in the unknown state");
+        }
+
+        vm.shutdown();
```

Comment:
I must be missing something but, when `delay` is 0, how is this different from what the `shutdown_timer` did?

Reply:
`shutdown_timer` requires a valid ssh_session, which we are not going to get if we can't determine the IP address of the instance.  The `UNKNOWN` status indicates we were unable to get an ssh session into to the instance to validate it was up and running.  See `wait_until_ssh_up`.

So, to be able to still stop the instance, we just force it here.

Reply:
I believe I got that same issue in states other than `UNKNOWN` (e.g. `STARTING`). IIUC, `VirtualMachine::shutdown` is also what shuts the machine down in the timer, the ssh session being used only to print wall messages. 

Wouldn't it then be better then to keep the `VM::shutdown` call in a single place but make the ssh session optional, rather than making the shutdown procedure depend on the state?

Reply:
Ok, changed:)

Reply:
Yeah, something along those lines is what I had in mind. It may be a bit more work but I thought it could save us hunting down state-dependent code later on. 

Of course there may be disadvantages I am not thinking of, so please do counter if you think that's the best! :)

Reply:
LOL ok thanks, crossed messages here, I hadn't seen your last when I replied